### PR TITLE
[API Deprecation] Deprecate node and edge Dataloader in dgl.dataloading

### PR DESCRIPTION
**Pull request by peizhou001**
## Description

Remove class NodeDataloader and EdgeDataloader in dgl.dataloading.

## Checklist
Please feel free to remove inapplicable items for your PR.
- [x] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [ ] I've leverage the [tools](https://docs.google.com/document/d/1iHyj7zlmygKSk5gBPsqIqL5ASPzJSPREaNT_QdsiYA4/edit) to beautify the python and c++ code.
- [ ] The PR is complete and small, read the [Google eng practice (CL equals to PR)](https://google.github.io/eng-practices/review/developer/small-cls.html) to understand more about small PR. In DGL, we consider PRs with less than 200 lines of core code change are small (example, test and documentation could be exempted).
- [x] All changes have test coverage
- [ ] Code is well-documented
- [x] To the best of my knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR
- [ ] If the PR is for a new model/paper, I've updated the example index [here](../examples/README.md).

### CODE REVIEW

**Comment by mufeili** [comment]
This must get approved by @BarclayII .

**Comment by BarclayII** [comment]
Currently the deprecation is blocked by the TGN example, which I think needs non-trivial work to rewrite its data loading pipeline to fit our new DataLoader API.  To unblock development and testing, given that TGN is not one of our golden examples, could we temporarily move the `_dataloading` folder to the TGN example, and make "refactor TGN" a low-priority work item?  @frozenbugs @jermainewang

**Comment by BarclayII** [comment]
Conclusion is that we are going to remove TGN example temporarily and add it back after refactoring post DGL 1.0 @peizhou001

**Comment by peizhou001** [comment]
> Conclusion is that we are going to remove TGN example temporarily and add it back after refactoring post DGL 1.0. @peizhou001 are you going to remove them and `_dataloading` in this PR?

No, _datalaoding and TGN deprecation is in #5043 , this PR concentrate on the NodeDataLoader and EddeDataloader deprecation in dgl.dataloading module.

File: examples/pytorch/gas/main_sampling.py
Comment (frozenbugs) [inline examples/pytorch/gas/main_sampling.py:54]: poorly aligned

File: examples/pytorch/gas/main_sampling.py
Comment (frozenbugs) [inline examples/pytorch/gas/main_sampling.py:68]: ditto.

File: examples/pytorch/tgn/dataloading.py
Comment (frozenbugs) [inline examples/pytorch/tgn/dataloading.py:279]: nit: avoid using \ to change newline. Please update since this PR touched this line :)

File: examples/pytorch/gas/main_sampling.py
Comment (peizhou001) [reply examples/pytorch/gas/main_sampling.py:54]: fixd

File: examples/pytorch/gas/main_sampling.py
Comment (peizhou001) [reply examples/pytorch/gas/main_sampling.py:68]: fixed

File: examples/pytorch/tgn/dataloading.py
Comment (peizhou001) [reply examples/pytorch/tgn/dataloading.py:279]: sure, changed for fix.

**Review by frozenbugs — APPROVED** [review]

**Review by BarclayII — APPROVED** [review]

**Pull request merged**
